### PR TITLE
[PLAT-11285] Add Navigation span attributes

### PR DESCRIPTION
**Pull request by gingerbenw**
## Goal

To add required attributes for navigation spans

## Design

As the public `startNavigationSpan` method returns a span without any way to add attributes, this PR refactors the react navigation plugin to accept the span factory as part of it's configuration method, using all internal functionality to start and end spans, and add new attributes.

## Testing

Update existing unit tests to add assertions for new span attributes

### CODE REVIEW

File: packages/core/lib/core.ts [hunk 1]
```diff
@@ -101,11 +101,11 @@ export function createClient<S extends CoreSchema, C extends Configuration, T> (
       })
 
       for (const plugin of configuration.plugins) {
-        plugins.push(plugin)
+        plugins.push(plugin as unknown as Plugin<C>)
```
Comment (gingerbenw) [inline packages/core/lib/core.ts:104]: little bit of a hack here I would like to see improved.

File: packages/react-navigation/test/navigation-context.test.tsx [hunk 2]
```diff
@@ -1,163 +1,150 @@
...
-import { InMemoryDelivery, VALID_API_KEY, createTestClient } from '@bugsnag/js-performance-test-utilities'
-import type BugsnagPerformance from '@bugsnag/react-native-performance'
+import { type SpanFactory } from '@bugsnag/core-performance'
+import { MockSpanFactory } from '@bugsnag/js-performance-test-utilities'
+import { type ReactNativeConfiguration } from '@bugsnag/react-native-performance'
 import { fireEvent, render, screen } from '@testing-library/react-native'
 import React, { useContext } from 'react'
 import { Button, View } from 'react-native'
 import { NavigationContext, NavigationContextProvider } from '../lib/navigation-context'
-import reactNativePlatformExtensions from './utils/react-native-platform-extensions'
```
Comment (yousif-bugsnag) [inline packages/react-navigation/test/navigation-context.test.tsx:8]: Can this file be deleted now it's no longer being used?

File: packages/react-navigation/test/navigation-context.test.tsx [hunk 3]
```diff
@@ -1,163 +1,150 @@
-import { type Span } from '@bugsnag/core-performance'
-import { InMemoryDelivery, VALID_API_KEY, createTestClient } from '@bugsnag/js-performance-test-utilities'
-import type BugsnagPerformance from '@bugsnag/react-native-performance'
```
Comment (yousif-bugsnag) [inline packages/react-navigation/test/navigation-context.test.tsx:3]: Do we still need a `BugsnagPerformance` mock or can that be removed now?

Comment (gingerbenw) [reply packages/react-navigation/test/navigation-context.test.tsx:3]: 💥

File: packages/react-navigation/test/navigation-context.test.tsx (hunk 2 above)
Comment (gingerbenw) [reply packages/react-navigation/test/navigation-context.test.tsx:8]: 💥

**Review by yousif-bugsnag — APPROVED** [review]

**Pull request closed without merging**